### DISCUSSION TITLE
bucket file uploading?

### DISCUSSION THREAD

**Discussion by tryggs** [supabase/supabase#7573] [Questions] [answered]
I am super confused as to how I am supposed to upload a file to a storage bucket.

I figured that [this](https://supabase.com/docs/reference/javascript/storage-from-upload) would help me but I couldn't figure out what is what. I find that that is a common issue with the supabase docs (which I still love) but sometimes it's just code and I don't understand what's supposed to go where.

anyway, I have a bucket called creatorbucket and a variable that is the file path to be uploaded, how would that piece of code look in javascript?

**Comment by Hallidayo** [comment, marked as answer]
Hi @tryggs,

Are you using a particular framework in your project or vanilla javascript?

If you are using a framework such as next.js then you could use the following which is used in this [example quickstart](https://supabase.com/docs/guides/with-nextjs#bonus-profile-photos).

The example shows how you can upload a file using the supabase .upload from function and also download a file using the supabase .download function.
